isic-archive-tasks/isic_archive_tasks/image.py

```diff
@@ -0,0 +1,199 @@
+    image = Image().load(imageId, force=True)
+
+    if image['ingested']:
+        logger.warning('Attempting to ingest an already ingested image.')
+        return
+
+    try:
+        imageData = Image().imageData(image)
+        image['meta']['acquisition']['pixelsY'] = imageData.shape[0]
+        image['meta']['acquisition']['pixelsX'] = imageData.shape[1]
+        image['readable'] = True
+        image['ingested'] = False
+        image['ingestionState'] = {
+            'large-image': None,
+            'superpixel-mask': None
+        }
+        Image().save(image)
+
+        callback = markImageIngested.s(image['_id'])
+        chord([generateSuperpixels.s(image['_id']),
+               generateLargeImage.s(image['_id'])])(callback)
+    except Exception:
+        sentry_sdk.capture_exception()
+        image['readable'] = False
+        image['ingested'] = True
+        Image().save(image)
+
+
+@app.task(base=CredentialedGirderTask)
+def markImageIngested(results, imageId):
+    from girder.plugins.isic_archive.models.image import Image
+    image = Image().load(imageId, force=True)
+    image['ingested'] = True
+    Image().save(image)
+
+
+@app.task(bind=True, base=CredentialedGirderTask)
+def generateSuperpixels(self, imageId):
+    try:
+        from girder.plugins.isic_archive.models.image import Image
+        from girder.plugins.isic_archive.models.segmentation_helpers.scikit import \
+            ScikitSegmentationHelper
+        image = Image().load(imageId, force=True)
+        imageFile = Image().originalFile(image)
+
+        originalFileStreamResponse = requests.get(
+            'http://isic-archive.test/api/v1/file/%s/download' % imageFile['_id'], headers={
+                'Girder-Token': self.token
+            })
+        originalFileStreamResponse.raise_for_status()
+        originalFileStreamResponse = io.BytesIO(originalFileStreamResponse.content)
+
+        # Scikit-Image is ~70ms faster at decoding image data
+        originalImageData = ScikitSegmentationHelper.loadImage(originalFileStreamResponse)
+
+        superpixelsData = ScikitSegmentationHelper.superpixels(originalImageData)
+        superpixelsEncodedStream = ScikitSegmentationHelper.writeImage(
+            superpixelsData, 'png')
+
+        uploadSuperpixelsResponse = requests.post('http://isic-archive.test/api/v1/file', params={
+            'parentType': 'item',
+            'parentId': imageId,
+            'name': '%s_superpixels_v%s.png' % (image['name'], SUPERPIXEL_VERSION),
+            'size': len(superpixelsEncodedStream.getvalue())
+        }, data=superpixelsEncodedStream.getvalue(), headers={
+            'Girder-Token': self.token
+        })
+        uploadSuperpixelsResponse.raise_for_status()
+
+        superpixelsFile = File().load(uploadSuperpixelsResponse.json()['_id'], force=True)
+        image = Image().load(imageId, force=True, exc=False)
+
+        superpixelsFileNameMatch = re.match(
```

Comment:
Remove this match and conditional. It was left over from the event handler, which was called for every file upload, and we needed to filter non-superpixel files.